shell/platform/linux/BUILD.gn

```diff
@@ -114,21 +114,45 @@ test_fixtures("flutter_linux_fixtures") {
 }
 
 executable("flutter_linux_unittests") {
+  public = _public_headers
   testonly = true
 
   sources = [
+    "fl_basic_message_channel.cc",
```

Comment:
Test targets shouldn't be building implementation files, they should be getting implementations via target dependencies. What was the driver for this?

Reply:
I was having the issue that the following didn't work:
```
deps += [ ":flutter_linux" ]
```
Because that links against the actual Flutter engine library, and collides with my mock engine.

I've had another go at rearranging the build rules but no idea if this is the right way to do it, please check again!